internal/common/tree.go

```diff
@@ -343,11 +446,11 @@ func findChildren(name string, record *v1alpha1.DNSRecord) []*DNSTreeNode {
 	return nodes
 }
 
-func findDataSets(name string, record *v1alpha1.DNSRecord) []DNSTreeNodeData {
-	dataSets := []DNSTreeNodeData{}
-	for _, ep := range record.Spec.Endpoints {
+func findDataSets(name string, endpoints []*endpoint.Endpoint) []*DNSTreeNodeData {
+	dataSets := []*DNSTreeNodeData{}
+	for _, ep := range endpoints {
 		if ep.DNSName == name {
-			dataSets = append(dataSets, DNSTreeNodeData{
+			dataSets = append(dataSets, &DNSTreeNodeData{
 				RecordType:       ep.RecordType,
 				RecordTTL:        ep.RecordTTL,
 				SetIdentifier:    ep.SetIdentifier,
```

Comment:
I'm not 100% i understand all this code, but why doesn't it just keep a pointer to Endpoint resources? Why does it have it's own `DNSTreeNodeData` struct?

Reply:
because the same host (node, e.g. klb.domain.com) can have multiple sets of targets with unique labels. In the DNS Record this is acheived with multiple endpoints with the same host, in the tree the host == node so the different endpoints (e.g. GEO * and GEO eu) are shown as 2 datasets for the same host.